packages/dashboards/src/lib/components/timeseries-widget/chart-presets/xy-chart/xy-chart.component.html

```diff
@@ -4,28 +4,77 @@
         <nui-chart class="w-100" [chart]="chartAssist.chart"></nui-chart>
     </div>
     <div class="d-flex" *ngIf="hasLegend()"
-         [ngClass]="[legendShouldBeAlignedRight() ? 'legend scrollable' : 'legend-bottom']">
+                [ngClass]="[legendShouldBeAlignedRight() ? 'legend scrollable' : 'legend-bottom']">
         <nui-legend class="m-auto"
                     [active]="chartAssist.isLegendActive"
                     [interactive]="true"
                     [orientation]="legendShouldBeAlignedRight() ? 'vertical' : 'horizontal'">
-            <nui-legend-series *ngFor="let legendSeries of chartAssist.legendSeriesSet; trackBy: chartAssist.seriesTrackByFn"
-                               [descriptionPrimary]="legendSeries.legendDescriptionPrimary"
-                               [descriptionSecondary]="legendSeries.legendDescriptionSecondary"
-                               [isSelected]="!chartAssist.isSeriesHidden(legendSeries.id)"
-                               (isSelectedChange)="chartAssist.toggleSeries(legendSeries.id, $event)"
-                               (mouseenter)="chartAssist.emphasizeSeries(legendSeries.id)"
-                               (mouseleave)="chartAssist.resetVisibleSeries()"
-                               [seriesRenderState]="chartAssist.renderStatesIndex[legendSeries.id]?.state"
-                               (click)="onLegendClick(legendSeries, $event)">
+            <nui-legend-series class="legend-container"
+                        [ngClass]="['state-' + chartAssist.renderStatesIndex[legendSeries.id]?.state]"
```

Comment:
it seems like this should already be taken care of if you use the seriesRenderState input:
https://github.com/solarwinds/nova/blob/0a504b1c5c55cafc5bd02e2845e6f0da1791f591/packages/charts/src/legend/legend-series/legend-series.component.html#L2